cpp-ch/local-engine/local_engine_jni.cpp

```diff
@@ -166,11 +166,6 @@ JNIEXPORT jint JNI_OnLoad(JavaVM * vm, void * /*reserved*/)
     return JNI_VERSION_1_8;
 }
 
-JNIEXPORT void JNI_OnUnload(JavaVM * vm, void * /*reserved*/)
-{
-    // manually destroy native in 'nativeDestroyNative' method
-}
```

Comment:
IIUC, in CH, the unloading actually does nothing?

Then why was the unloading code from Java side strictly required for CH backend in current code base? https://github.com/apache/incubator-gluten/blob/fdd83714067c4a77006d6695b2bd8f6ad9e3ef23/backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHListenerApi.scala#L84-L89

Reply:
No, we actually need unloading. We curently move to `Java_org_apache_gluten_vectorized_ExpressionEvaluatorJniWrapper_nativeDestroyNative` see https://github.com/apache/incubator-gluten/pull/9242

The reason why we do this is https://github.com/openjdk/jdk17u-dev/commit/75a8b7fa831ec22a149f271c119a2496c1c5ee87  doesn't allow unload jni for built-in classloader

![img_v3_02kv_a7e0c14c-d3e5-49d3-817e-09826dfa3ccg](https://github.com/user-attachments/assets/9975bbf9-1fea-479f-af03-ccd25563a904)

Reply:
Great. I think we can remove the Java-side unloading API then?